python/ray/dashboard/client/src/pages/dashboard/state.ts

```diff
@@ -83,5 +87,55 @@ const slice = createSlice({
   },
 });
 
+const clusterWorkerPids = (
+  rayletInfo: RayletInfoResponse,
+): Map<string, Set<number>> => {
+  // Groups PIDs registered with the raylet by node IP address
+  // This is used to filter out processes belonging to other ray clusters.
+  const nodeMap = new Map();
+  const workerPids = new Set();
+  for (const [nodeIp, { workersStats }] of Object.entries(rayletInfo.nodes)) {
+    for (const worker of workersStats) {
+      if (!worker.isDriver) {
+        workerPids.add(worker.pid);
+      }
+    }
+    nodeMap.set(nodeIp, workerPids);
+  }
+  return nodeMap;
+};
+
+const filterNonClusterWorkerInfo = (
+  rayletInfo: RayletInfoResponse,
+  nodeInfo: NodeInfoResponse,
+) => {
+  // The back-end that generates the NodeInfoResponse does not remove worker
```

Comment:
Isn't it already implemented by you in the past? What's the reason why we have this logic again?

Reply:
I moved the logic out of NodeInfo.tsx for a couple reasons:
1. That file was getting very long and noisy because of the additional feature type defs/imports, grouping logic, and sorting logic, and I wanted to simplify it
2. On looking at it further it made more sense to me to do this at the redux level, because then none of the front-end components have to worry about workers from other clusters. I'm trying to fix the problem as far upstream as possible (even better would be the front-end, but it's a little more work).